File: DSA/Ayush_Mishra_590014844/Day_52_Question_2.c
#include <stdio.h>
#include <stdlib.h>

#define MAX 9999

int main() {
    int n, k;
    printf("Enter size of array: ");
    scanf("%d", &n);
    int arr[n], hash[MAX] = {0};
    printf("Enter array elements: ");
    for (int i = 0; i < n; i++) {
        scanf("%d", &arr[i]);
        hash[arr[i]] = 1;
    }
    printf("Enter difference value k: ");
    scanf("%d", &k);
    printf("Pairs with difference %d:\n", k);
    for (int i = 0; i < n; i++) {
        if (arr[i] + k < MAX && hash[arr[i] + k])
            printf("(%d, %d)\n", arr[i], arr[i] + k);
        if (arr[i] - k >= 0 && hash[arr[i] - k])
            printf("(%d, %d)\n", arr[i], arr[i] - k);
    }
    return 0;
}

File: DSA/Ayush_Mishra_590014844/Day_52_Question_2.py
def find_pairs_with_difference(arr, k):
    seen = set(arr)
    pairs = []
    for num in arr:
        if num + k in seen:
            pairs.append((num, num + k))
        if num - k in seen:
            pairs.append((num, num - k))
    return pairs

def main():
    n = int(input("Enter size of array: "))
    arr = list(map(int, input("Enter array elements: ").split()))
    k = int(input("Enter difference value k: "))
    result = find_pairs_with_difference(arr, k)
    print("Pairs with difference", k, ":")
    for pair in result:
        print(pair)

if __name__ == "__main__":
    main()
